drivers/hid/hid-multitouch.c

```diff
@@ -1673,6 +1694,57 @@ static void mt_expired_timeout(struct timer_list *t)
 	clear_bit(MT_IO_FLAGS_RUNNING, &td->mt_io_flags);
 }
 
+static void update_keyboard_backlight(struct hid_device *hdev, bool enabled)
+{
+	struct usb_device *udev = hid_to_usb_dev(hdev);
+	unsigned char *buffer;
+	int ret;
+
+	/* Wake up the device in case it's already suspended */
+	pm_runtime_get_sync(&udev->dev);
+
+	buffer = kzalloc(MS_TYPE_COVER_FEATURE_REPORT_SIZE, GFP_KERNEL);
+	if (!buffer) {
+		hid_err(hdev, "Failed to send Type Cover hid report, couldn't allocate buffer\n");
+		goto out;
+	}
+
+	buffer[0] = MS_TYPE_COVER_FEATURE_REPORT_ID;
+	buffer[1] = 0xff;
+	buffer[2] = 0x00;
+	buffer[3] = 0xff;
+	buffer[4] = enabled ? 0x01 : 0x00;
+
+	ret = hid_hw_raw_request(hdev, MS_TYPE_COVER_FEATURE_REPORT_ID,
+				 buffer, MS_TYPE_COVER_FEATURE_REPORT_SIZE,
+				 HID_FEATURE_REPORT, HID_REQ_SET_REPORT);
+	kfree(buffer);
+
+	if (ret != MS_TYPE_COVER_FEATURE_REPORT_SIZE)
+		hid_err(hdev, "Failed to send Type Cover hid report: %d\n", ret);
+
+out:
+	pm_runtime_put_sync(&udev->dev);
```

Comment:
I think you don't need the `_sync` variant here, the async one should work fine as well.

Reply:
Hmm yeah, theoretically the kernel should be able to handle the usb device being resumed in the `suspend` callback, in practice this sounds like something that's rarely tested. Given that the device has to be suspended anyway, I'd prefer doing it here right away which makes sure we leave the callback in the same state as when we entered it...